contracts/Core/StakeManager.sol

```diff
@@ -183,30 +183,27 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         require(sToken.balanceOf(msg.sender) >= lock.amount, "locked amount lost"); // Can Use ResetLock
 
         uint256 rAmount = _convertSRZRToRZR(lock.amount, staker.stake, sToken.totalSupply());
-        require(sToken.burn(msg.sender, lock.amount), "Token burn Failed");
+        uint256 lockAmount = lock.amount;
         staker.stake = staker.stake - rAmount;
-
         // Function to Reset the lock
         _resetLock(stakerId);
-
+        require(sToken.burn(msg.sender, lockAmount), "Token burn Failed");
         // Transfer commission in case of delegators
         // Check commission rate >0
         if (stakerIds[msg.sender] != stakerId && staker.commission > 0) {
             uint256 commission = (rAmount * staker.commission) / 100;
-            require(razor.transfer(staker._address, commission), "couldnt transfer");
             rAmount = rAmount - commission;
+            require(razor.transfer(staker._address, commission), "couldnt transfer");
         }
 
         //Transfer stake
         require(razor.transfer(msg.sender, rAmount), "couldnt transfer");
-
-        emit Withdrew(epoch, stakerId, rAmount, staker.stake, block.timestamp);
```

Comment:
This event you have removed, but not added, please add so

Reply:
yes I will add

Reply:
fixed

Reply:
fixed